test/jdk/tools/jlink/plugins/CDSPluginTest.java

```diff
@@ -87,8 +90,24 @@ public static void main(String[] args) throws Throwable {
        if (Platform.isLinux()) {
            System.out.println("---- Test different platforms scenario ----");
            String jlinkPath = JDKToolFinder.getJDKTool("jlink");
-           String[] cmd = {jlinkPath, "--add-modules", "java.base,java.logging",
-                           "-J-Dos.name=windows", "--generate-cds-archive",
+           // copy over the java.base and java.logging module file to a temporary directory
+           // which we then use as a --module-path during jlink image generation. using such a
+           // separate --module-path will force the JLink task to read the ModuleTarget from
+           // the java.base module-info.class to identify the target platform for the image
+           // being generated.
+           Path jdkRoot = Path.of(jlinkPath).getParent().getParent();
```

Comment:
Or simply get from `test.jdk` system property:

```suggestion
          Path jdkRoot = Path.of(System.getProperty("test.jdk"));
```

Reply:
Hello Mandy, the `JDKToolFinder.getJDKTool("jlink")` uses the `test.jdk` to try and find the tool and then falls back to another system property, if the tool wasn't found. That's why I had used this `getParent()` technique to locate the JDK root. But I think using just the `test.jdk` system property should be fine too, since as far as I can see in the test definition of this test, there isn't anything specific that would require the test JDK to be present in any different place than the `test.jdk` system property. I've updated the PR to use this suggested approach.